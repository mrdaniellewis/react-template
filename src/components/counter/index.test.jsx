import React from 'react';
import Counter from '.';
import { INCREMENT } from '../../actions';

it('matches snapshot', () => {
  expect(mountWithStore(<Counter />)).toMatchSnapshot();
});

it('clicking button calls increment', () => {
  const component = mountWithStore(<Counter />);
  const spy = jest.spyOn(component.instance().store, 'dispatch');
  component.update();
  component.find('button').simulate('click');
  expect(spy).toHaveBeenCalledWith({ type: INCREMENT });
});

it('shows current current count', () => {
  const component = mountWithStore(<Counter />, { count: 1 });
  expect(component).toMatchSnapshot();
});
