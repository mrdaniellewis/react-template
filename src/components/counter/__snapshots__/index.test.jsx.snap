// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increase count
    0
  </button>
</div>
`;

exports[`shows current current count 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increase count
    1
  </button>
</div>
`;
